packages/backend/src/tokens/request.ts

```diff
@@ -147,6 +155,14 @@ function assertProxyUrlOrDomain(proxyUrlOrDomain: string | undefined) {
   }
 }
 
+function handleRequestStateForApiRoutes(options: AuthenticateRequestOptions, requestState: RequestState): RequestState {
```

Comment:
@dimkl Let's add unit tests for this scenario

Reply:
I will add unit tests when the API is finalized :)